src/main/java/strategy/handlers/PurchaseHandlerImpl.java

```diff
@@ -0,0 +1,15 @@
+package strategy.handlers;
+
+import db.Storage;
+import model.FruitTransaction;
+import strategy.OperationHandler;
+
+public class PurchaseHandlerImpl implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction fruitTransaction) {
+        String fruitName = fruitTransaction.getFruitName();
+        Integer storageFruitQuanity = Storage.fruitsStorage.get(fruitName);
+        int transactionQuantity = fruitTransaction.getQuantity();
+        Storage.fruitsStorage.replace(fruitName, storageFruitQuanity - transactionQuantity);
```

Comment:
add exception if quantity is negative number

Reply:
done